extensions/amp-analytics/0.1/requests.js

```diff
@@ -220,20 +303,21 @@ export class RequestHandler {
       }
     }
     return Promise.all(requestPromises).then(() => {
-      return this.getExtraUrlParamsString_(params);
+      return params;
     });
   }
 
   /**
    * Handle the params map and form the final extraUrlParams string
    * @param {!Object} params
+   * @param {boolean} opt_includeNull
    * @return {string}
    */
-  getExtraUrlParamsString_(params) {
+  getExtraUrlParamsString_(params, opt_includeNull) {
     const s = [];
     for (const k in params) {
       const v = params[k];
-      if (v == null) {
+      if (v == null && !opt_includeNull) {
```

Comment:
I'm wondering in what case `null` is needed

Reply:
v could be `null` in two cases. 1. user assign a var to null, 2. filter return null (not in prod and to be refactored)

We remove `param=null` by default. But since batchin-plugin function is taking control, I figure it's best they take care of null param. 

I am open to remove null param anyway. let me know

Reply:
For the record, richmedia individually would never need a parameter value of null. Perhaps you can expect anyone needing it to use the string 'null' rather than an actual null?

Reply:
I think we can keep it simple for now and add it back when someone need it.

Reply:
SGTM